x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearnToRankService.java

```diff
@@ -176,15 +179,25 @@ private QueryExtractorBuilder applyParams(QueryExtractorBuilder queryExtractorBu
             return queryExtractorBuilder;
         }
 
-        Script script = new Script(ScriptType.INLINE, DEFAULT_TEMPLATE_LANG, templateSource, Collections.emptyMap());
-        String parsedTemplate = scriptService.compile(script, TemplateScript.CONTEXT).newInstance(params).execute();
-        // TODO: handle missing params.
-        XContentParser parser = XContentType.JSON.xContent().createParser(parserConfiguration, parsedTemplate);
-
-        return new QueryExtractorBuilder(
-            queryExtractorBuilder.featureName(),
-            QueryProvider.fromXContent(parser, false, INFERENCE_CONFIG_QUERY_BAD_FORMAT)
-        );
+        try {
+            Script script = new Script(ScriptType.INLINE, DEFAULT_TEMPLATE_LANG, templateSource, SCRIPT_OPTIONS, Collections.emptyMap());
+            String parsedTemplate = scriptService.compile(script, TemplateScript.CONTEXT).newInstance(params).execute();
+            XContentParser parser = XContentType.JSON.xContent().createParser(parserConfiguration, parsedTemplate);
+
+            return new QueryExtractorBuilder(
+                queryExtractorBuilder.featureName(),
+                QueryProvider.fromXContent(parser, false, INFERENCE_CONFIG_QUERY_BAD_FORMAT)
+            );
+        } catch (GeneralScriptException e) {
+            if (e.getRootCause() instanceof MustacheInvalidParameterException == false) {
```

Comment:
@benwtrent Is it possible that the instanceof is return misleading result because of a different classloader for the mustache module?

I have:

```
e.getRootCause() instanceof MustacheInvalidParameterException == false
```

and 

```
e,getRootCause().getClass().getName().equals(MustacheInvalidParameterException.class.getName()) == true
```

Reply:
Bonus question: what is the cleanest way to write comparison in that case?

Reply:
@afoucret If you're genuinely curious you should be able to compare the class loaders from `e.getRootCause()` and `MustacheInvalidParameterException` to confirm this theory. If I remember correctly, we do have to do some magic to get all the class loaders to work correctly with plugins.